src/Illuminate/Foundation/Support/Providers/EventServiceProvider.php

```diff
@@ -48,4 +51,58 @@ public function listens()
     {
         return $this->listen;
     }
+
+    /**
+     * Get the discovered events and listeners for the application.
+     *
+     * @return array
+     */
+    protected function discoveredEvents()
+    {
+        if ($this->app->eventsAreCached()) {
+            return require $this->app->getCachedEventsPath();
+        }
+
+        return $this->shouldDiscoverEvents()
+                    ? $this->discoverEvents()
+                    : [];
+    }
+
+    /**
+     * Determine if events and listeners should be automatically discovered.
+     *
+     * @return bool
+     */
+    public function shouldDiscoverEvents()
+    {
+        return false;
+    }
+
+    /**
+     * Discover the events and listeners for the application.
+     *
+     * @return array
+     */
+    public function discoverEvents()
+    {
+        return collect($this->discoverEventsWithin())
+                    ->reduce(function ($discovered, $directory) {
+                        return array_merge(
+                            $discovered,
+                            DiscoverEvents::within($directory, base_path())
+                        );
+                    }, []);
+    }
+
+    /**
+     * Get the listener directories that should be used to discover events.
+     *
+     * @return array
+     */
+    protected function discoverEventsWithin()
+    {
+        return [
+            app_path('Listeners'),
```

Comment:
Since we have app in the service provider can we use `$this->app->basePath('Listeners')` instead of the helper here?